platformio/util.py

```diff
@@ -51,7 +51,7 @@ def get_lib_dir():
         config = get_project_config()
         if (config.has_section("platformio") and
                 config.has_option("platformio", "lib_dir")):
-            lib_dir = config.get("platformio", "lib_dir")
+            lib_dir = abspath(config.get("platformio", "lib_dir"))
```

Comment:
Need to rewrite this block of code. `abspath` doesn't work with `~` home signs. In this situation, `lib_dir` will be incorrect for the value `lib_dir = ~/Documents/Arduino/Libraries`.
1. Need to expand user's path
2. Then we can convert relative path to absolute. 

Can you change it?

P.S: Thanks a lot for your contribution! :+1: